### ISSUE TITLE
Monster classification and organization.

### ISSUE THREAD

**Issue by calculuschild** [AdventureLookup/AdventureLookup#201]
The Common Monsters list is _yuuuuge_. Would there be any benefit to creating a more limited list of monsters with general types? So instead of "Skeletons, Undead, Zombie, Ju Ju Zombie, Ogre Zombie, Beholder Zombie, Aquatic Zombie, Centaur Zombie, Mummy, Mummy Lord, Dread Mummy, Wight, Wraith, Lich, Demilich, Dracolich, Lich Hound, Revenant, etc. etc. etc.," there would be a box for "Undead." I think first of all, there would be more people looking for adventures featuring "undead" than specifically an "aquatic zombie".

Maybe if this were to happen, it would be possible that next to the "undead" checkbox, there is a little drop-down arrow opening up a sub-list of "Zombies, Mummies, Liches, Other" or something, and if even those weren't good enough, those could have sub-lists.

Classifying and grouping related monsters provides an additional benefit of revealing to the DM, "you wanted Zombies? Well right here I have all the different zombies in a convenient place for you, and if you've never heard of Aquatic Zombies before, now's your chance because they aren't buried in a list of hundreds of random creatures!" If an entry is obscure enough, hiding it in a huge list just makes it useless because of two issues: 1) Nobody knows it exists in the first place. 2) Nobody will realistically stumble upon it by chance. In both cases it is not used and just creates clutter.

The whole point of this site is to make is easy for DMs to find adventures with the content they want. Having to sift through literally hundreds of monsters does not accomplish that goal.

**Comment by cmfcmf** [comment]
This might be related to #137.

**Comment by MattColville** [comment]
I think the general rule should be "Juju zombie" minimum, as long as JuJu Zombie, is also a Zombie and also Undead."

Right now, our tools are pretty primitive, but if we could have a system whereby a curator could select "All JuJu Zombies" and tag them as "Undead" without having to do it by hand, we'd be good.

Also, it would be nice if the really heavily populated fields had their own little search bar so people had an option other than "expand this list into oblivion."

**Comment by calculuschild** [comment]
I don't think we need any fancy tools for a curator to start sorting these monsters. All adventures that have JuJu Zombies should be referring to the same JuJu Zombie entry. We only need to classify that one entry and it should apply everywhere. No need to select "All JuJu Zombies" since there is only one.

So, let me see if this sounds reasonable as a workflow:

- Member A fills in an adventure that involves JuJu Zombies, sees that such a monster doesn't exist yet, and adds it as a brand new monster.
- Curator B checks up on a list that shows all unclassified monsters, and notices JuJu Zombies has been added as a new monster and isn't classified yet. Until something happens, JuJu Zombies will just show up as its own unclassified entry in the search panel.
- Curator B does some research and discovers that JuJu Zombies are, in fact, a sub-type of Zombies, and marks them as such. Zombies at this point have already been classified as a sub-type of Undead, so JuJu Zombies are now "Undead" -> "Zombie" -> "JuJu Zombie".
- Member C comes along and adds another adventure that also uses JuJu Zombies. It has already been categorized, so when he adds it to the adventure, it will already fall under "Undead" -> "Zombie" -> "JuJu Zombie"
- Dungeon Master D comes along and is looking for an adventure involving zombies. He looks through the monster list and sees a listing for "Undead" with a little expansion arrow next to it. He decides "Undead" isn't specific enough, so he expands the list to see "Zombies, Skeletons, Liches, Vampires, etc.," each with their own expansion arrows. If he were to expand the Zombies category, he would now see "Aquatic Zombies, JuJu Zombies, Zombie Horseman, Zombie Mathematician, etc." However, he decides "Zombie" is specific enough, and checks the box for Zombies and applies the search. He opens the first adventure in the results and looks through the info. This happens to be the JuJu Zombies adventure Member A filled in.
- Dungeon Master D looks through the adventure info and sees JuJu Zombies in the list of monsters in the adventure. **How should that appear?** Does it just show "JuJu Zombie," or should it be "JuJu Zombie (Undead - Zombie)", or does he see "JuJu Zombie," "Zombie," and "Undead" all as separate monsters?
- Now, Member E adds a new adventure and simply puts "Zombie" as the monster type.  Zombie is already classified as "Undead," so all is well. **When D looks at this adventure, does he just see "Zombie (Undead)?"**
- Member F adds a new adventure and just puts "Undead" as the monster type. **Do we flag this as too general and notify a curator? Or is this fine?**

There's obviously a lot that would go into this, but I think the task of beginning the sorting could be started pretty soon. Maybe even into a spreadsheet until the system supports it.

**Comment by marcellarius** [comment]
Having dealt with this in other projects, a natural approach is a tree, like @calculuschild suggests. If adventures reference the monster taxonomy, you to build tools to curate the tree (rename, move and merge monsters) in bulk.

If you use a closure table in the database, it's very easy to do subtree queries so Filter by: Zombie could match Ash Zombies, Juju Zombies, etc. 

The real world rarely fits tidily into boxes, and this approach means you'll end up with cases where a monster fits into multiple parts of the hierarchy. A lacedon is both undead and aquatic.
